src/configure/configure.ts

```diff
@@ -516,36 +514,49 @@ class Orchestrator {
         }
         this.inputs.pipelineConfiguration.template = this.inputs.potentialTemplates[0];
 
-        //If RepoAnalysis is disabled or didn't provided response related to language of selected template
-        this.inputs.repositoryAnalysisApplicationSettings = new RepositoryAnalysisApplicationSettings();
-
-        //Post selecting the template update this.inputs.repositoryAnalysisApplicationSettings with corresponding languageSettings
-        if (!!repoAnalysisResult
-            && !!repoAnalysisResult.repositoryAnalysisApplicationSettingsList) {
-
-            //Get languageSettings (corresponding to language of selected settings) provided by RepoAnalysis
-            await this.updateRepositoryAnalysisApplicationSettings(repoAnalysisResult);
-        }
         telemetryHelper.setTelemetry(TelemetryKeys.ChosenTemplate, this.inputs.pipelineConfiguration.template.label);
-
     }
 
     private async updateRepositoryAnalysisApplicationSettings(repoAnalysisResult: RepositoryAnalysisParameters): Promise<void> {
-        var applicationSettings = repoAnalysisResult.repositoryAnalysisApplicationSettingsList.filter(applicationSetting => {
+        //If RepoAnalysis is disabled or didn't provided response related to language of selected template
+        this.inputs.repositoryAnalysisApplicationSettings = new RepositoryAnalysisApplicationSettings();
+
+        if (!repoAnalysisResult || !repoAnalysisResult.applicationSettingsList) {
+            return;
+        }
+        var applicationSettings = repoAnalysisResult.applicationSettingsList.filter(applicationSetting => {
+            if (this.inputs.pipelineConfiguration.template.templateType === TemplateType.remote) {
+                const template = this.inputs.pipelineConfiguration.template as RemotePipelineTemplate;
+                if (applicationSetting.language && applicationSetting.language !== template.attributes.language.toLowerCase()) {
```

Comment:
Rather than finding out applicationSettings by comparing these properties, a better way is to use workingDirectory which is returned by template service (may be store WD in RemotePIpelineTemplate object if not already doing so. cc @tejasd1990 for helping here

Reply:
I will add workingDirectory check also. But to do more filtering of repositoryAnalysisSettings we need to compare these properties as well. Consider a scenarios where a workingDirectory contains two types of application settings ex: gruntFile settings and DockerFile settings and suppose user has selected DockerFile template. In this case as the workingDirectory will be same for gruntFile settings and DockerFile settings, we won't be able to filter on workingDirectory alone, we will have to filter by comparing application setting (language, deployTarget and buildTarget).